ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/validation/BlobSidecarGossipValidator.java

```diff
@@ -241,19 +256,63 @@ public SafeFuture<InternalValidationResult> validate(final BlobSidecar blobSidec
                 return reject("BlobSidecar block header signature is invalid.");
               }
 
-              if (!receivedValidBlobSidecarInfoSet.add(
-                  new SlotProposerIndexAndBlobIndex(
-                      blockHeader.getSlot(),
-                      blockHeader.getProposerIndex(),
-                      blobSidecar.getIndex()))) {
-                return ignore(
-                    "BlobSidecar is not the first valid for its slot and index. It will be dropped.");
+              synchronized (this) {
```

Comment:
why synchronised when using `LimitedSet.createSynchronized` ?

Reply:
"It is imperative that the user manually synchronize on the returned map when traversing any of its collection views via Iterator, Spliterator or Stream:"
Looks like we are safe on atomic operations, removed